packages/hdwallet-keepkey/src/utils.ts

```diff
@@ -62,3 +62,32 @@ export function translateOutputScriptType(scriptType: core.BTCOutputScriptType):
   }
   throw new Error("unhandled OutputScriptType enum: " + scriptType);
 }
+
+//codereview.stackexchange.com/questions/265820/using-javascript-given-a-json-value-recursively-find-all-json-objects-then-so
+const strSorter = (a: any, b: any) => (a > b ? 1 : a < b ? -1 : 0);
+const isObj = (obj: any) => "object" === typeof obj && null !== obj;
+
+const sortInPlaceObjectKeys = (obj: any) => {
+  const sorted = Object.entries(obj).sort((a, b) => strSorter(a[0], b[0]));
+  for (const [key] of sorted) {
+    delete obj[key];
+  }
+  return Object.assign(obj, Object.fromEntries(sorted));
+};
+
+export function sortTxFields(obj: any): any {
```

Comment:
just curious if this was also a req for legacy amino on osmosis, or just to make tx payloads match keplr references?

Reply:
I've done this here so that we've got a canonical format for all transactions at the time of signing - recursively sorted, snake_case. This resolves any ambiguity about what is actually being signed over and matches the Keplr unsigned transaction payloads as well.

Reply:
worth hoisting this util and using it for all tendermint txs to keep things normalized across wallets/chains?